lib/alex_koin/slack_commands.ex

```diff
@@ -1,10 +1,18 @@
 defmodule AlexKoin.SlackCommands do
+  require Ecto.Query
   alias AlexKoin.Repo
+  alias AlexKoin.Account
   alias AlexKoin.Account.{User, Wallet, Transaction}
   alias AlexKoin.Coins.Coin
 
-  def get_or_create(slack_id) do
-    case User |> Repo.get_by(slack_id: slack_id) do
+  # Use slack to populate name and email
+  def get_or_create(slack_id, slack) do
+    user_info = case Map.fetch(slack.users, slack_id) do
+      :error -> nil
+      {:ok, info} -> info
+    end
+
+    user = case User |> Repo.get_by(slack_id: slack_id) do
```

Comment:
Don't know if I've seen a pipe in a case statement like this before. Again, would probably separate this into standalone function.